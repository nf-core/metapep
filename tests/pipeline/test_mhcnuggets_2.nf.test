nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_mhcnuggets_2"
    tag "pipeline"

    test("test mhcnuggets hla type 2 prediction") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },

                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml").exists() },

                { assert snapshot(path("$outputDir/db_tables/alleles.tsv"),
                                path("$outputDir/db_tables/conditions_alleles.tsv"),
                                path("$outputDir/db_tables/conditions.tsv"),
                                path("$outputDir/db_tables/entities_proteins.tsv"),
                                path("$outputDir/db_tables/entities.tsv"),
                                path("$outputDir/db_tables/microbiomes_entities.no_weights.tsv"),
                                path("$outputDir/db_tables/microbiomes_entities.nucl.tsv"),
                                path("$outputDir/db_tables/microbiomes_entities.tsv"),
                                path("$outputDir/db_tables/microbiomes.tsv"),
                                path("$outputDir/db_tables/proteins_peptides.tsv"),
                                path("$outputDir/db_tables/peptides.tsv.gz"),
                                path("$outputDir/db_tables/proteins.tsv.gz"),
                                path("$outputDir/db_tables/stats.txt")).match("db_tables") },
                { assert new File("$outputDir/db_tables/predictions.tsv.gz").exists() },

                { assert new File("$outputDir/figures/entity_binding_ratios.DPB1_01_01.pdf").exists() },
                { assert new File("$outputDir/figures/entity_binding_ratios.with_points.DPB1_01_01.pdf").exists() },
                { assert new File("$outputDir/figures/prediction_score_distribution.DPB1_01_01.pdf").exists() },

                { assert snapshot(path("$outputDir/logs/prediction_warnings.log")).match("logs") },

                { assert new File("$outputDir/prodigal/proteins.pred_0.tsv.gz").exists() }
            )
        }

    }

}
