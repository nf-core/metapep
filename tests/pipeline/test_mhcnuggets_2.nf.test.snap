{
    "db_tables": {
        "content": [
            "alleles.tsv:md5,ab0e4dbbb55b8154d37256018c19d67c",
            "conditions_alleles.tsv:md5,769c299e94c7df654a5a14da2771c95d",
            "conditions.tsv:md5,a6fe5daaf11a90b61c83c9934969c52c",
            "entities_proteins.tsv:md5,2ba821a04755841dc6f116965f853f40",
            "entities.tsv:md5,61f364c97c883f45663ae27d115b229f",
            "microbiomes_entities.no_weights.tsv:md5,bd449dafeb86c374552f2e22a9780b19",
            "microbiomes_entities.nucl.tsv:md5,3888070b634251c0ae3e9fd4857140d7",
            "microbiomes_entities.tsv:md5,6cef150e7f5fc2f3320d9a93da00f5de",
            "microbiomes.tsv:md5,3d597aec25752f7ca83436b31bfaa128",
            "proteins_peptides.tsv:md5,c2a16b2d30c9cf5ea52207dd799b4603",
            "peptides.tsv.gz:md5,dd0f94bea5079df8ef06895de255038b",
            "proteins.tsv.gz:md5,104b786dfa89e183a46e29f1f37f17de",
            "predictions.tsv.gz:md5,d88dfc746b74451b1cd00eb06ea834bb",
            "stats.tsv:md5,8da7e6e2d9802fe424f09eeae7bcf98e"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-20T16:30:33.586676622"
    },
    "logs": {
        "content": [
            "prediction_warnings.log:md5,d3ff6ac2a68d620e8f5023f4ab16a1b9"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-26T08:14:54+0000"
    }
}